packages/core/src/modules/connections/services/ConnectionService.ts

```diff
@@ -642,6 +642,28 @@ export class ConnectionService {
     return connectionRecord
   }
 
+  public async addConnectionType(agentContext: AgentContext, connectionRecord: ConnectionRecord, type: string) {
+    const tags = (connectionRecord.getTag('connectionType') as string[]) || ([] as string[])
```

Comment:
Now that the typing is correctly set on getTags, I think the casting is not needed anymore?

Reply:
If we use `getTag('connectionType')` it's still needed. However I think it's more elegant to use `getTags().connectionType`, where the typing is already there as you say.